src/hooks/useValidateRepository.ts

```diff
@@ -15,16 +34,25 @@ const branchSearchString = 'branch:';
 const pathSearchString = 'path:';
 const endSearchString = ', context';
 
-const getErrorMessage = (rawErrorString: string, searchString: string) =>
-  rawErrorString.slice(
+const getErrorMessage = (rawErrorString: string, searchString: string) => {
+  return rawErrorString.slice(
     rawErrorString.indexOf(searchString) + searchString.length,
     rawErrorString.indexOf(endSearchString)
   );
+};
 
 const useValidateRepository = (
   getFieldValue: (name: NamePath) => string,
-  setValidationState: React.Dispatch<any>,
-  validateRepository: any
+  setValidationState: React.Dispatch<React.SetStateAction<ValidationState>>,
+  validateRepository: MutationTrigger<
```

Comment:
can we simplify this type? worst case, I would still prefer using () => void instead of this

Reply:
It's important as `validateRepository` is a `UseMutationHook`. It's not a function in general meaning.

Reply:
Actually, for the `useValidateRepository` it's only important that it's a function that takes a validation payload and returns promise with data or error